make.py

```
@@ -122,12 +122,43 @@ def build(target: str):
+
+    with open(containerfile, "r") as f:
+        pattern = re.compile(r"\$\{([^}:]+)(?::-(.+?))?\}")
+        # TODO handle line continuations
+        lines = [
+            x
+            for x in f.read().splitlines()
+            if x.startswith("FROM ") or x.startswith("ARG ")
+        ]
+        for base_image in [x[5:].strip() for x in lines if x.startswith("FROM ")]:
+            if " AS " in base_image.upper():
+                base_image = re.split(" AS ", base_image, 1, flags=re.IGNORECASE)[0]
```

Ruff (0.14.4)

`re.split` should pass `maxsplit` and `flags` as keyword arguments to avoid confusion due to unintuitive argument positions

(B034)